# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1682959137 -7200
# Parent  68d00867f551e6262f10d4ec6109aa951918628e
Feature: pref mail.biff.show_for_server to disable biff display in tree for servers.

diff --git a/mailnews/base/src/FolderUtils.jsm b/mailnews/base/src/FolderUtils.jsm
--- a/mailnews/base/src/FolderUtils.jsm
+++ b/mailnews/base/src/FolderUtils.jsm
@@ -76,26 +76,29 @@ function getFolderProperties(aFolder, aO
 
   properties.push("folderNameCol");
 
   properties.push("serverType-" + aFolder.server.type);
 
   // set the SpecialFolder attribute
   properties.push("specialFolder-" + getSpecialFolderString(aFolder));
 
-  // Now set the biffState
-  switch (aFolder.biffState) {
-    case nsIMsgFolder.nsMsgBiffState_NewMail:
-      properties.push("biffState-NewMail");
-      break;
-    case nsIMsgFolder.nsMsgBiffState_NoMail:
-      properties.push("biffState-NoMail");
-      break;
-    default:
-      properties.push("biffState-UnknownMail");
+  if (aFolder.isServer &&
+      Services.prefs.getBoolPref("mail.biff.show_for_server", true)) {
+    // Now set the biffState
+    switch (aFolder.biffState) {
+      case nsIMsgFolder.nsMsgBiffState_NewMail:
+        properties.push("biffState-NewMail");
+        break;
+      case nsIMsgFolder.nsMsgBiffState_NoMail:
+        properties.push("biffState-NoMail");
+        break;
+      default:
+        properties.push("biffState-UnknownMail");
+    }
   }
 
   properties.push("isSecure-" + aFolder.server.isSecure);
 
   // A folder has new messages, or a closed folder or any subfolder has new messages.
   if (
     aFolder.hasNewMessages ||
     (!aOpen && aFolder.hasSubFolders && aFolder.hasFolderOrSubfolderNewMessages)
diff --git a/mailnews/mailnews.js b/mailnews/mailnews.js
--- a/mailnews/mailnews.js
+++ b/mailnews/mailnews.js
@@ -872,16 +872,18 @@ pref("mail.biff.on_new_window", true);
 // messages, as per the classic Thunderbird definition. Defaults to false, which
 // notifies about the number of unread messages.
 pref("mail.biff.use_new_count_in_badge", false);
 #endif
 #ifdef XP_WIN
 pref("mail.biff.use_new_count_in_badge", true);
 #endif
 
+pref("mail.biff.show_for_server", true);
+
 // For feed account serverType=rss sound on biff; if true, mail.biff.play_sound.* settings are used.
 pref("mail.feed.play_sound", false);
 
 // Content disposition for attachments (except binary files and vcards).
 //   0= Content-Disposition: inline
 //   1= Content-Disposition: attachment
 pref("mail.content_disposition_type", 1);
 
